### PR TITLE
Add support for python 3.5

### PR DESCRIPTION
**Pull request by GuyHoozdis**
I added python3.5 to the `envlist` in `tox.ini` and updated the project so that all the tests continued to pass under all the configurations in the `envlist`.

```
$ tox

... snip ...

OK
__________________________________________________________________________________
summary
__________________________________________________________________________________
  py27-django15: commands succeeded
  py27-django16: commands succeeded
  py27-django17: commands succeeded
  py27-django18: commands succeeded
  py27-djangorq: commands succeeded
  py35-django18: commands succeeded
  congratulations :)
```

### CODE REVIEW

**Comment by crccheck** [comment]
Can you update the classifiers in setup.py to have:

`Programming Language :: Python :: 2.7`
`Programming Language :: Python :: 3.5`

File: canary_endpoint/resources/databases.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def check(self):
         result = super(Database, self).check()
         try:
             cursor = self.connection.cursor()
-            map(cursor.execute, self.statements)
+            [cursor.execute(sql) for sql in self.statements]
```
Comment (nosamanuel) [inline canary_endpoint/resources/databases.py]: How about a `for` loop instead? Executing as a by-product of a list comprehension looks like you are creating a variable that never gets assigned to anything (expected: `statements = [transform(sql) for sql in self.statements]`).

File: tests/test_endpoint.py [hunk 2]
```diff
@@ -14,12 +14,12 @@ def setUp(self):
         self.mock_ok_elasticsearch()
         self.expected_data = self.get_fixture('ok.json')
 
-    def assert_content(self, content, status='ok', **overrides):
-        data = json.loads(content)
+    def assert_response_content(self, response, status='ok', **overrides):
+        encoding = getattr(response, 'charset', 'utf-8')
```
Comment (nosamanuel) [inline tests/test_endpoint.py]: We're supporting two different behaviors: one for UTF-8 and one for who-knows-what. Should we use a concrete value instead so we know the exact behavior?

File: tests/test_service_resources.py [hunk 3]
```diff
@@ -40,7 +40,12 @@ def test_error_result_reports_error_on_success(self):
...
+        json_decoding_error_messages = {
+            'python2_error': 'No JSON object could be decoded',
+            'python3_error': 'Expecting value: line 1 column 1 (char 0)',
+        }
+
         result = self.service.check()
         self.assertEqual(result['status'], ERROR)
         self.assertEqual(result['result'], None)
-        self.assertEqual(result['error'], 'No JSON object could be decoded')
+        self.assertIn(result['error'], json_decoding_error_messages.values())
```
Comment (nosamanuel) [inline tests/test_service_resources.py]: I've recently been sold on the idea that tests should [expect literals](http://blog.jayfields.com/2008/02/testing-expect-literals.html). If we handle the decode error in the service canary as a special case, we can guarantee a consistent error message across Python versions, right?

File: tests/test_endpoint.py (hunk 2 above)
Comment (GuyHoozdis) [inline tests/test_endpoint.py]: I checked the [JSON Specification](https://tools.ietf.org/html/rfc4627#section-3) before I wrote this code...

> JSON text SHALL be encoded in Unicode.  The default encoding is UTF-8.

...but really that line of code is about supporting **different versions of Django** not different encodings.  Older versions of Django will not have the `charset` attribute on the response object.  In that case, we use the default value.

---

You are correct; I could have done something like this:

``` python
def assert_content(self, content, status='ok', **overrides):
    data = json.loads(content.decode('utf-8'))
    # ... snip ...
```

I could certainly do that.  What do you think?

File: tests/test_service_resources.py (hunk 3 above)
Comment (GuyHoozdis) [inline tests/test_service_resources.py]: > handle the decode error [..] as a special case, we can guarantee a consistent error message [..], right?
> I'm glad you point that out!  I had considered doing that, but this was simpler and so I started here.  I think you are right; a consistent error message across Python versions would be preferable.

---

That was an interesting [article](http://blog.jayfields.com/2008/02/testing-expect-literals.html) you shared.

> To be clear, the expected value should be the literal itself, not a variable holding a literal that was created earlier in the test.

I have been of the opposite mind, but I think that he make some good points.  I'll give it a try for a while and see if I think it makes my life and my code better.

File: tests/test_endpoint.py (hunk 2 above)
Comment (nosamanuel) [inline tests/test_endpoint.py]: 👍 less contingent, more concrete.

**Comment by GuyHoozdis** [comment]
I think I hit all the comments from review.  Let me know if there is anything else.

**Comment by nosamanuel** [comment]
👍

**Pull request merged**